todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java

```diff
@@ -77,20 +77,26 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View root = inflater.inflate(R.layout.taskdetail_frag, container, false);
         setHasOptionsMenu(true);
+
+        // Retain the fragment to persist the presenter and its state.
+        // This means that it won't be recreated on a configuration change, but
+        // it's dangerous as it can produce bugs and leaks. Use it wisely.
+        setRetainInstance(true);
```

Comment:
what do you think about using the `Controller` to deal with the lifecycle? Retaining the `Fragment` just to hold a reference to the `Presenter` (which is not View related) seems like a bad idea